### PR TITLE
addon icon and background.js code

### PR DESCRIPTION
**Pull request by RaviShawnJha**

### CODE REVIEW

File: addon/background-script.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/*console.log(document.querySelectorAll('iframe'))*/
+/* setTimeout(() => {
+  document.querySelector('iframe').src = "https://icatcare.org/app/uploads/2018/06/Layer-1704-1920x840.jpg"
+}, 1000) */
+let ad = document.querySelectorAll(iframe)
```
Comment (nbriz) [inline addon/background-script.js]: querySelectorAll expects arguments to be in strings, so this should be:
```js
 let ad = document.querySelectorAll('iframe')
```

File: addon/background-script.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+for (let i = 0; i < ad.length; i++ ){
+  let father = ad[i].parentElement
+father.removeChild(ad[i])
+const img =
+father.appendChild('img.src = https://icatcare.org/app/uploads/2018/06/Layer-1704-1920x840.jpg')
```
Comment (nbriz) [inline addon/background-script.js]: appendChild expects an actual element object as it's argument, not a string like this. so first u need to create an img element
```js
const img = document.createElement('img')
```
then u can change that img element's src:
```js
img.src = 'https://icatcare.org/app/uploads/2018/06/Layer-1704-1920x840.jpg'
```
&& only then can u append it to the parent element
```js
father.appendChild(img)
```

**Review by nbriz — CHANGES_REQUESTED** [review]
did u test this code? i tried it but it didn't work for me i got a few errors (see my comments)

**Comment by nbriz** [comment]
@RaviShawnJha so typically i like to leave comments on specific lines of code that need fixing, but there's a lot of tiny things that are off here, so instead it might be more helpful if i explain what u need to have here && why...

first u need to use `querySelectorAll` to create a list of all the iframes on a page like this:
```js
let ad = document.querySelectorAll('iframe')
```
in ur code u have a `{` at the end of this line, but that isn't necessary, u aren't _defining_ a function (that's when u'd have the `{ ... }` to create a section for the internal logic), u're _executing_ or using a function that already exists (namely querySelectorAll)

once u've got a list of iframes (which in ur case u've stored in a variable called `ad`, which means `ad[0]` would be the first iframe on the page, `ad[1]` would be the second, etc.) u can create a single for loop which will loop as many times as their are ads on the page, by using the `ad.length` (which returns the number of elements inside the `ad` list)

```js
for (let i = 0; i < ad.length; i++ ){
  // whatever u want to do to or with each ad goes here.
  // ...
  // a simple test to make sure this is working could be to
  // log the src (urls) of each of the iframes like this:
  let url = ad[i].src
  console.log(url)
}
```
u can try that in ur console && u should see a list of URLs get logged (one URL for each iframe, aka ad, on the page). that's great for testing, but what u actually want to do inside of the for loop is the following:
1. find the parent element of each ad (the element the iframe is inside of)
```js
let father = ad[i].parentElement
```
2. then remove the iframe from that parent element
```js
father.removeChild(ad[i])
```
3. then create a new image element
```js
const img = document.createElement('img')
```
4. then set that img element's `src` property to the image u want (this is supposed to be the images that others upload to ur group's server, but for now we'll just hard code a cat gif)
```js
img.src = 'https://i.imgur.com/lVlPvCB.gif'
```
5. then put that image inside the element that used to have the ad (iframe) in it.
```js
father.appendChild(img)
```
agian, all of that should be inside of the for loop, b/c this is all stuff u want to do to each ad

**Comment by RaviShawnJha** [comment]
Thank you. That worked! - I had not gotten before that I needed to declare img inside the for loop!
see my new code in this request - Will I now be able to replace the last bit of code with lines that assign img.src = 'the url which contains uploaded images' and then father.appendChild(img[z]).

File: addon/background-script.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+let ad = document.querySelectorAll('iframe')
+
+for (let i = 0; i < ad.length; i++ ){
+  let father = ad[i].parentElement
+  father.removeChild(ad[i])
+
+const img = document.createElement('img')
+
+  for (let z = 0; z < images.length; z++ ){
```
Comment (nbriz) [inline addon/background-script.js:9]: what are u trying to do with this second for loop? there is no variable called `images` so `images.length` is undefined, which means this loop won't actually do anything (other than potentially cause a bug), unless i'm misunderstanding something u're trying to do (let me know if that's the case) i think u should remove this (as well as it's closing `}` on line 12. the 2 lines inside this loop.

**Review by nbriz — CHANGES_REQUESTED** [review]
this is nearly perfect, but there's still some unnecessary code

File: addon/background-script.js (hunk 3 above)
Comment (RaviShawnJha) [reply addon/background-script.js:9]: I intended to use this loop to put different images from the array of uploaded images into the places where the ads were.

Comment (nbriz) [reply addon/background-script.js:9]: i see. the problem is that array doesn't exist, u need to request it from ur server's API using the `fetch()` function (that's how u request data from APIs) i'll explain in more detail below.

**Pull request closed without merging**